compiler/rustc_middle/src/lint.rs

```diff
@@ -246,6 +251,9 @@ pub fn struct_lint_level<'s, 'd>(
         let has_future_breakage =
             future_incompatible.map_or(false, |incompat| incompat.future_breakage.is_some());
 
+        let is_force_warn = matches!(level, Level::ForceWarn)
+            || matches!(src, LintLevelSource::CommandLine(_, Level::ForceWarn));
```

Comment:
Why do we need the `if` here?

Reply:
Oh, I guess because the level could be *error*? Maybe add a comment about that

Reply:
I'm not sure what you mean (this line doesn't have an `if` in it). What in particular do you not find clear?

Reply:
Oh, I meant the `||` -- like, why isn't it sufficient to just check if the level is `ForceWarn`?

Reply:
sorry, in my brain, `||` and `if` are equivalent

Reply:
haha no worries (I think if you would have said "conditional" it would have registered, but the use of backticks made it seem like you were talking about a literal `if` in the code). 

The level can be lower (e.g., warn or allow) because `--cap-lints` globally lints caps. The interaction between the global options `--cap-lints` and the per lint options `--force-warn` is a tiny bit hacky IMO, but I don't think refactoring the entire system is worth it.  A comment here should make it sufficiently clear.

Reply:
Actually, this is false. I think check of the src can be removed.